### PR TITLE
src: support --disable_result_log and --enable_result_log

### PR DESCRIPTION
**Pull request by Defined2014**

### CODE REVIEW

File: src/main.go [hunk 1]
```diff
@@ -109,6 +109,10 @@ type tester struct {
 	// use --disable_query_log or --enable_query_log to control it
 	enableQueryLog bool
 
+	// enable result log will output to result file or not.
+	// use --enable_result_log or --disable_result_log to control it
+	enableResultLog bool
```
Comment (bb7133) [inline src/main.go:114]: I don't quite get what does 'result log' mean? Could you provide an example? Thanks.

Comment (Defined2014) [reply src/main.go:114]: An example:

The test
```
--disable_result_log
show tables;
--enable_result_log
show tables;
```

The result
```
show tables;
show tables;
Tables_in_show
```

Comment (Defined2014) [reply src/main.go:114]: Ref https://github.com/mysql/mysql-server/blob/ea1efa9822d81044b726aab20c857d5e1b7e046a/client/mysqltest.cc#L557

File: src/main.go [hunk 2]
```diff
@@ -869,6 +878,10 @@ func (t *tester) executeStmt(query string) error {
 			return errors.Trace(err)
 		}
 
+		if !t.enableResultLog {
+			return nil
+		}
+
```
Comment (tiancaiamao) [inline src/main.go:884]: How about move to after dumpToByteRows?
I'm not sure is whether the sql.Row a lazy object, if so, we may just ignore error that may happen during really use.

File: src/main.go (hunk 1 above)
Comment (bb7133) [reply src/main.go:114]: Got it! Thanks.

**Review by bb7133 — APPROVED** [review]
LGTM

File: src/main.go (hunk 2 above)
Comment (Defined2014) [reply src/main.go]: Done

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**